src/lang/pt/mod.rs

```diff
@@ -0,0 +1,433 @@
+        assert_invalid!("sessenta quinze");
+        assert_invalid!("sessenta cem");
+        assert_invalid!("sessenta quatro");
+        assert_invalid!("cem e um");
+        assert_invalid!("cento mil");
+    }
+
+    #[test]
+    fn test_zeroes() {
+        assert_text2digits!("zero", "0");
+        assert_text2digits!("zero oito", "08");
+        assert_text2digits!("zero um", "01");
+        assert_text2digits!("zero uma", "01");
+        assert_text2digits!("zero zero cento e vinte e cinco", "00125");
+        assert_invalid!("cinco zero");
+        assert_invalid!("cinquenta zero três");
+        assert_invalid!("cinquenta e zero três");
+        assert_invalid!("cinquenta e zero");
+        assert_invalid!("cinquenta e três zero");
+        assert_invalid!("dez zero");
+    }
+
+    #[test]
+    fn test_ordinals() {
+        assert_text2digits!("vigésimo quarto", "24º");
+        assert_text2digits!("vigésimo primeiro", "21º");
+        assert_text2digits!("centésimo primeiro", "101º");
+        assert_text2digits!("décima sexta", "16ª");
+        assert_text2digits!("décimas sextas", "16ᵃˢ");
+        assert_text2digits!("décimos sextos", "16ᵒˢ");
+    }
+
+    #[test]
+    fn test_replace_numbers_integers() {
+        assert_replace_numbers!(
+            "vinte e cinco vacas, doze galinhas e cento e vinte e cinco kg de batatas.",
+            "25 vacas, 12 galinhas e 125 kg de batatas."
+        );
+        assert_replace_numbers!("mil duzentos e sessenta e seis dólares.", "1266 dólares.");
+        assert_replace_numbers!("um dois três quatro vinte quinze.", "1 2 3 4 20 15.");
+        assert_replace_numbers!(
+            "um, dois, três, quatro, vinte, quinze.",
+            "1, 2, 3, 4, 20, 15."
+        );
+        assert_replace_numbers!("um dois três quatro trinta e cinco.", "1 2 3 4 35.");
+        assert_replace_numbers!("vinte e um, trinta e um.", "21, 31.");
+        assert_replace_numbers!("trinta e quatro ≠ trinta quatro", "34 ≠ 30 4");
+        assert_replace_numbers!("cem e dois", "100 e 2");
+    }
+
+    #[test]
+    fn test_replace_numbers_formal() {
+        assert_replace_numbers!(
+            "trinta e três nove sessenta zero seis doze vinte e um",
+            "33 9 60 06 12 21"
+        );
+        assert_replace_numbers!(
+            "zero nove sessenta zero seis doze vinte e um",
+            "09 60 06 12 21"
+        );
+    }
+
+    #[test]
+    fn test_replace_numbers_that_use_conjunction() {
+        assert_replace_numbers!("sessenta seis", "60 6");
+        assert_replace_numbers!("sessenta e seis", "66");
+        assert_replace_numbers!("duzentos e quarenta e quatro", "244");
+        assert_replace_numbers!("dois mil e vinte", "2020");
+        assert_replace_numbers!("mil novecentos e oitenta e quatro", "1984");
+        assert_replace_numbers!("mil e novecentos", "1900");
+        // assert_replace_numbers!(
+        //     "mil novecentos",
+        //     "1000 900"
+        // );
```

Comment:
TODO

Reply:
Ex.:
- mil noventos et seis => 1906
- mil e noventos => 1900
- mil noventos, incorrect,  so => 1000 900

We would need to introduce some look-ahead in the framework.
In practice, if the speaker really means "1000 900", they would put some silence between the two words and we take than into account when processing transcriptions.

In the longer term, we'll need look-ahead for sure.